TestProject/src/rules/BaseRule.php

```diff
@@ -0,0 +1,39 @@
+<?php
+namespace GameOfLife\Rules;
+
+use GameOfLife\options\Getopt;
+use GameOfLife\cellularAutomat\Field;
+
+/**
+ * API-class for additional rules.
```

Comment:
API-class is not a good name in this case.
Better description would be: `Base Rule for all rule implementations`.
For baseclasses it is also important to explain what a developer should do when implementing it. This is also helpful in the method documentation of the methods that should be implemented.